elasticsearch/bulk/bulk.go

```diff
@@ -180,11 +185,47 @@ func (b *Bulk) flushMessages() error {
 func (b *Bulk) bulkRequest() error {
 	startedTime := time.Now()
 	reader := bytes.NewReader(b.batch)
-	_, err := b.esClient.Bulk(reader)
+	resp, err := b.esClient.Bulk(reader)
 	b.metric.BulkRequestProcessLatencyMs = time.Since(startedTime).Milliseconds()
 	if err != nil {
 		return err
 	}
+	err = b.handleResp(resp)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (b *Bulk) handleResp(resp *esapi.Response) error {
+	if resp == nil {
+		return fmt.Errorf("ERROR occurred when bulk request. Error: Response is nil")
+	}
+	if resp.IsError() {
+		return fmt.Errorf("ERROR occurred when bulk request. Error:%s", resp.String())
+	}
+	bodyBuffer := new(bytes.Buffer)
+	_, err := bodyBuffer.ReadFrom(resp.Body)
+	if err != nil {
+		return err
+	}
+	bodyInterface := make(map[string]interface{})
+	err = json.Unmarshal(bodyBuffer.Bytes(), &bodyInterface)
```

Comment:
we are generally using github.com/json-iterator/go. we could use it here again maybe

Reply:
I changed the library with github.com/json-iterator/go.